src/core/ddsc/tests/psmx.c

```diff
@@ -1264,6 +1326,378 @@ CU_Test (ddsc_psmx, basic)
   dds_delete (dds_get_parent (participant));
 }
 
+/// @brief Check that shared memory can be enabled and the used locator can be set.
+/// @methodology
+/// Case with Iceoryx:
+/// - Check that the data types I'm planning to use are actually suitable for use with shared memory.
+/// - Expectation: They are memcopy-safe.
+/// - Create a configuration with a psmx interface capable of shared memory and specify the locator.
+/// - Create a domain using this configuration.
+/// - Create some entities
+/// - Check the locator used for shared memory.
+/// - Expectation: The locator is the same as specified in the config for the domain.
+/// - Check if shared memory is enabled.
+/// - Expectation: Shared memory is enabled.
+/// - Write and read some topics.
+/// - Expectation: Sent data is received by readers.
+/// 
+/// - Create a configuration with a psmx interface capable of shared memory and don't specify a locator.
+/// - Create a domain using this configuration.
+/// - Create some entities
+/// - Check the locator used for shared memory.
+/// - Expectation: The second half of the locator is all zeros.
+/// - Check if shared memory is enabled.
+/// - Expectation: Shared memory is enabled.
+/// 
+/// Case with cdds:
+/// - Create a configuration with a psmx interface incapable of shared memory.
+/// - Create a domain using this configuration.
+/// - Create some entities
+/// - Check if shared memory is enabled.
+/// - Expectation: Shared memory is not enabled.
+///
+CU_Test(ddsc_psmx, shared_memory)
+{
+  #define sample_cnt 8
+  {
+    // Check that the data types I'm planning to use are actually suitable for use with shared memory.
+    dds_data_type_properties_t props;
+    props = dds_stream_data_types(SC_Model_desc.m_ops);
+    CU_ASSERT_FATAL((props & DDS_DATA_TYPE_IS_MEMCPY_SAFE) == DDS_DATA_TYPE_IS_MEMCPY_SAFE);
+    props = dds_stream_data_types(PsmxType1_desc.m_ops);
+    CU_ASSERT_FATAL((props & DDS_DATA_TYPE_IS_MEMCPY_SAFE) == DDS_DATA_TYPE_IS_MEMCPY_SAFE);
+  }
+
+  const dds_domainid_t domainId = 42;
+  char* CDDS_PSMX_NAME = ddsrt_expand_envvars("${CDDS_PSMX_NAME:-cdds}", domainId);
+  bool using_psmx_iox = (strcmp(CDDS_PSMX_NAME, "iox") == 0);
+  ddsrt_free(CDDS_PSMX_NAME);
+
+  if ( using_psmx_iox ) {
```

Comment:
I checked it, but `dds_is_shared_memory_available()` internally uses `dds_psmx_supported_features()`. In other words, if I call `dds_psmx_supported_features()` first, it can't help but give the same result for `dds_is_shared_memory_available()`. So it seems a bit pointless to me to do it that way.

Reply:
What I meant is that the `else` block (from line 1636) can be completely removed (and so the `if` on line 1376) if you parameterize the "is shared memory available". That can be done by either hard-coding that the `iox` implementation uses shared memory, and `cdds` not, or by requesting the features from the plugin. The latter will also work if we decide to add more implementations to run this test on, so I think that's the way to go. Calling `dds_is_shared_memory_available()` on the endpoints could be used to check if the returned value is consistent with the features the PSMX instance claims to support.